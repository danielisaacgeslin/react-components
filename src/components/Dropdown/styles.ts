import { StyleSheet } from 'aphrodite/no-important';
import { Color } from 'utils/dist/models/general';
import { hexToRgba } from 'utils/dist/utils';

export default StyleSheet.create({
  container: {
    position: 'absolute',
    boxShadow: `0 1px 5px 0 ${hexToRgba(Color.BLACK, 20)}`,
    backgroundColor: Color.WHITE,
    zIndex: 200,
    borderRadius: 3,
    padding: '10px 15px',
    opacity: 0,
    color: Color.GRAY,
    fontSize: 15
  },
  open: {
    opacity: 1
  },
  small: {
    borderRadius: 100,
    padding: '7px 10px'
  }
});
