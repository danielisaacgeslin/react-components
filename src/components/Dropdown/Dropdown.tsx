import React, { memo, useCallback } from 'react';
import { css } from 'aphrodite/no-important';

import styles from './styles';

export interface IDropdownProps {
  isOpen: boolean;
  top?: number;
  bottom?: number;
  left?: number;
  right?: number;
  small?: boolean;
}

const Dropdown = ({ isOpen, top, bottom, left, right, small, children }: React.PropsWithChildren<IDropdownProps>) => {
  const handleClick = useCallback((event: React.MouseEvent) => event.stopPropagation(), []);

  return (
    <div
      key="Dropdown"
      className={css([styles.container, isOpen && styles.open, small && styles.small])}
      style={{ top, bottom, left, right }}
      onClick={handleClick}
    >
      {isOpen && children}
    </div>
  );
};

export default memo(Dropdown);
