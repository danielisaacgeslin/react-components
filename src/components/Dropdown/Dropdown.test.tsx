import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import Dropdown, { IDropdownProps } from './Dropdown';

describe('Dropdown Component', () => {
  let Component;
  let props: IDropdownProps;

  beforeEach(() => {
    props = {
      isOpen: true
    };
    Component = shallow(<Dropdown {...props} />);
  });

  it('should render closed', () => {
    props.isOpen = false;
    Component = shallow(<Dropdown {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render open', () => {
    props.isOpen = true;
    Component = shallow(<Dropdown {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render small', () => {
    props.small = true;
    Component = shallow(<Dropdown {...props} />);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should stop propagation', () => {
    const event = { stopPropagation: jest.fn() };
    Component.find('div')
      .first()
      .simulate('click', event);
    expect(event.stopPropagation).toBeCalled();
  });
});
