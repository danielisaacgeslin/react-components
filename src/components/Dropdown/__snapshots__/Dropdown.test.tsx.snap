// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component should render closed 1`] = `
.container {
  position: absolute;
  box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.2);
  background-color: #FFF;
  z-index: 200;
  border-radius: 3px;
  padding: 10px 15px;
  opacity: 0;
  color: #D5DEF1;
  font-size: 15px;
}

<div
  className="container"
  onClick={[Function]}
  style={
    Object {
      "bottom": undefined,
      "left": undefined,
      "right": undefined,
      "top": undefined,
    }
  }
/>
`;

exports[`Dropdown Component should render open 1`] = `
.container-open {
  position: absolute;
  box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.2);
  background-color: #FFF;
  z-index: 200;
  border-radius: 3px;
  padding: 10px 15px;
  color: #D5DEF1;
  font-size: 15px;
  opacity: 1;
}

<div
  className="container-open"
  onClick={[Function]}
  style={
    Object {
      "bottom": undefined,
      "left": undefined,
      "right": undefined,
      "top": undefined,
    }
  }
/>
`;

exports[`Dropdown Component should render small 1`] = `
.container-open-small {
  position: absolute;
  box-shadow: 0 1px 5px 0 rgba(0, 0, 0, 0.2);
  background-color: #FFF;
  z-index: 200;
  color: #D5DEF1;
  font-size: 15px;
  opacity: 1;
  border-radius: 100px;
  padding: 7px 10px;
}

<div
  className="container-open-small"
  onClick={[Function]}
  style={
    Object {
      "bottom": undefined,
      "left": undefined,
      "right": undefined,
      "top": undefined,
    }
  }
/>
`;
